pilot/pkg/bootstrap/server.go

```diff
@@ -588,7 +588,7 @@ func (s *Server) initSecureDiscoveryService(args *PilotArgs, port string) error
 
 	cfg := &tls.Config{
 		GetCertificate: s.getIstiodCertificate,
-		ClientAuth:     tls.VerifyClientCertIfGiven,
+		ClientAuth:     tls.NoClientCert,
```

Comment:
This would break the newly added authentication of XDS clients. Why do you need this ?

Reply:
We don't, its verifying against the self-signed root cert though currently not the kubernetes root cert. And the change required to update that everywhere was pretty invasive so just to see if tests passed, etc, for now I disabled it. We certainly will not if we merge this